preprocessing/parsers.py

```diff
@@ -2,17 +2,24 @@
 import os
 import re
 import shutil
+from typing import List
 
 import pydicom
 
 
-def load_scan(dirpath: str):
-    """Takes in the path of a directory containing scans and
-    returns a list of dicom dataset objects. Each dicom dataset
-    contains a single image slice.
+def load_scan(dirpath: str) -> List[pydicom.FileDataset]:
+    """Loads a CT scan.
+
+    This only loads dicom files ending in .dcm and ignores nested
+    folders.
+
+    :param dirpath: the path to a directory containing dicom (.dcm) files
+    :return: a list of all dicom FileDataset objects, sorted by
+        ImagePositionPatient
     """
     slices = [pydicom.read_file(dirpath + '/' + filename)
-              for filename in os.listdir(dirpath)]
+              for filename in os.listdir(dirpath)
+              if filename.endswith('.dcm')]
```

Comment:
Looks like invalid syntax

Reply:
This seems like a proper for-if comprehension to me: https://www.digitalocean.com/community/tutorials/understanding-list-comprehensions-in-python-3.